mapper.sh

```diff
@@ -88,6 +88,17 @@ if [ $xargs_exit -ne 0 ]; then
 	exit 1
 fi
 
+# TODO: configurability and error handling
+#~ downscale() {
+	#~ convert $1 -resize 50% $1
+#~ }
+
+downscale() {
```

Comment:
Optimally, there'd be three functions: `downscale_convert`, `downscale_perc` and `downscale`. The `downscale` function would then check whether the `downscaling/perc` file is present, and if it is, it'd invoke `downscale_perc`, otherwise it'd invoke `downscale_convert`. That's doable with `if -f downscaling/perc; then` etc.

Reply:
I've defined the downscale depending on the existence of downscaling/perc now.